me.gladwell.eclipse.m2e.android/src/main/java/me/gladwell/eclipse/m2e/android/configuration/classpath/ReorderClasspathConfigurer.java

```diff
@@ -0,0 +1,50 @@
+/*******************************************************************************
+ * Copyright (c) 2015 David Carver and Ricardo Gladwell
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *******************************************************************************/
+
+package me.gladwell.eclipse.m2e.android.configuration.classpath;
+
+import static java.util.Arrays.asList;
+
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+
+import me.gladwell.eclipse.m2e.android.configuration.Classpaths;
+import me.gladwell.eclipse.m2e.android.configuration.ProjectConfigurationException;
+import me.gladwell.eclipse.m2e.android.project.IDEAndroidProject;
+import me.gladwell.eclipse.m2e.android.project.MavenAndroidProject;
+
+import org.eclipse.core.runtime.NullProgressMonitor;
+import org.eclipse.jdt.core.IClasspathEntry;
+import org.eclipse.jdt.core.IJavaProject;
+import org.eclipse.jdt.core.JavaCore;
+import org.eclipse.jdt.core.JavaModelException;
+import org.eclipse.m2e.jdt.IClasspathDescriptor;
+
+public class ReorderClasspathConfigurer implements RawClasspathConfigurer {
+
+    public void configure(MavenAndroidProject mavenProject, IDEAndroidProject eclipseProject, IClasspathDescriptor classpath) {
+        try {
+            IJavaProject javaproject = JavaCore.create(eclipseProject.getProject());
+            IClasspathEntry[] tosort = javaproject.readRawClasspath();
+            IClasspathEntry[] sorted = sortClasspath(mavenProject, tosort);
+            javaproject.setRawClasspath(sorted, new NullProgressMonitor());
```

Comment:
I agree. However, I'm trying to experiment with alternatives, but the m2e `IClasspathDescriptor` class doesn't allow you to update all its entries, and is basically not easy to order. Instead you have to remove all entries and then add them again in the right order which is also ugly and seems to drop the `/gen` folder for some reason.

Unfortunately, the Java sort algorithms also only seem to work on arrays, not sure if there is some other way we could order it.

Any ideas?